lib/http/node.js

```diff
@@ -120,6 +126,31 @@ NodeHttpClient.prototype.getAgent = function (parsedUrl, proxy, mock) {
   } else {
     agent = getFromCacheOrCreate(HttpAgent, agentOptions, parsedUrl);
   }
+
+  // try to detect and log PROXY envvar
+  const envProxy = Util.getEnvProxy();
```

Comment:
is this check still run for each request? maybe we can convert this block of code into a method and run when creating a new agent inside `getFromCacheOrCreate`?

Reply:
actually a good idea, wish this approach occurred to me sooner :) 
rewrote the proxy detection function and put the functionality into the new agent creation step so hopefully it only gets executed once, when the new agent is created